core/trino-main/src/main/java/io/trino/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -237,6 +237,12 @@ public synchronized void startLifespan(Lifespan lifespan, ConnectorPartitionHand
         scheduleGroups.put(lifespan, new ScheduleGroup(partitionHandle));
         whenFinishedOrNewLifespanAdded.set(null);
         whenFinishedOrNewLifespanAdded = SettableFuture.create();
+
+        // schedule a task for collecting dynamic filters in case some dependent split generator is waiting for them
+        if (dynamicFilterService.isCollectingTaskNeeded(stageExecution.getStageId().getQueryId(), stageExecution.getFragment())) {
+            checkArgument(lifespan.isTaskWide(), "Dynamic filtering cannot be used with grouped execution");
```

Comment:
If this is a new restriction, make sure to document it.

Reply:
It's not new. We have one in `io/trino/sql/planner/LocalExecutionPlanner.java:2871` already